src/xrc/src/forex.rs

```diff
@@ -1861,14 +1883,63 @@ mod test {
                     base_asset_num_received_rates: 5,
                     quote_asset_num_queried_sources: 5,
                     quote_asset_num_received_rates: 5,
+                    forex_timestamp: Some(1234),
                 }
             },
         );
 
         assert_eq!(store.allocated_bytes(), 273);
     }
 
-    /// This functiont ests the the forexes can report the max response bytes needed
+    /// This function tests the "go back" mechanism where when there are not rates for a requested timestamp, we may go back up to [MAX_DAYS_TO_GO_BACK] days.
+    #[test]
+    fn forex_go_back_days() {
+        let mut store = ForexRateStore::new();
+
+        let timestamp = 1661990400; // Corresponds to 2022-09-01
+        let queried_timestamp = timestamp + SECONDS_PER_DAY * MAX_DAYS_TO_GO_BACK;
+
+        store.put(
+            timestamp,
+            hashmap! {
+                "EUR".to_string() => QueriedExchangeRate {
+                    base_asset: Asset {
+                        symbol: "EUR".to_string(),
+                        class: AssetClass::FiatCurrency,
+                    },
+                    quote_asset: Asset {
+                        symbol: USD.to_string(),
+                        class: AssetClass::FiatCurrency,
+                    },
+                    timestamp,
+                    rates: vec![10_000],
+                    base_asset_num_queried_sources: 5,
+                    base_asset_num_received_rates: 5,
+                    quote_asset_num_queried_sources: 5,
+                    quote_asset_num_received_rates: 5,
+                    forex_timestamp: Some(timestamp),
+                }
+            },
+        );
+
+        // Assert that we can retrieve rates up to [MAX_DAYS_TO_GO_BACK] days back.
+        assert_eq!(
+            store
+                .get(queried_timestamp, queried_timestamp, "EUR", USD)
+                .unwrap()
+                .forex_timestamp
+                .unwrap(),
+            timestamp
+        );
+        // But also that we cannot retrieve rates for more than that.
+        let queried_timestamp = queried_timestamp + SECONDS_PER_DAY;
```

Comment:
If it's one second less, it would still find the rate because the timestamps are "rounded" to the start of the day, right?

Reply:
Yes